subsys/net/lib/tftp/tftp_client.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 InnBlue
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Generic header files required by the TFTPC Module. */
+#include <zephyr.h>
+#include <errno.h>
+#include <net/socket.h>
+
+/* Defines for creating static arrays for TFTP communication. */
+#define TFTP_MAX_FILENAME_SIZE   CONFIG_TFTPC_MAX_FILENAME_SIZE
+#define TFTP_HEADER_SIZE         4
+#define TFTP_BLOCK_SIZE          512
+#define TFTP_MAX_MODE_SIZE       8
+#define TFTP_REQ_RETX            CONFIG_TFTPC_REQUEST_RETRANSMITS
+
+/* Maximum amount of data that can be received in a single go ! */
+#define TFTPC_MAX_BUF_SIZE       (TFTP_BLOCK_SIZE + TFTP_HEADER_SIZE)
+
+/* TFTP Default Server IP / Port. */
+#define TFTPC_DEF_SERVER_IP      CONFIG_TFTPC_DEFAULT_SERVER_IP
+#define TFTPC_DEF_SERVER_PORT    CONFIG_TFTPC_DEFAULT_SERVER_PORT
+
+/* TFTP Opcodes. */
+#define RRQ_OPCODE               0x1
+#define WRQ_OPCODE               0x2
+#define DATA_OPCODE              0x3
+#define ACK_OPCODE               0x4
+#define ERROR_OPCODE             0x5
+
+/* TFTP Client Success / Error Codes. */
+#define TFTPC_DATA_RX_SUCCESS     1
```

Comment:
Anyone having thoughts on these error codes ? Some of these are directly conveyed to the application user and so they should be moved to the public `tftp.h` file I suppose ? 

In general, is there any guidelines to add new error codes (to ensure no overlap) or I can add whatever number I want ?